config/config/src/test/java/io/helidon/config/ConfigMappersTest.java

```diff
@@ -242,6 +246,96 @@ public void testBuiltinMappersSubNodeProperties() throws MalformedURLException {
         assertThat(key2Properties, hasEntry("key2.key23", "value23"));
     }
 
+    @Test
+    void testBuiltinEnumMapper() {
+        ConfigMapperManager manager = BuilderImpl.buildMappers(ConfigMapperManager.MapperProviders.create());
+        Config config = createEnumConfig();
+        MyType type = manager.map(config.get("type-a"), MyType.class);
+        assertThat("Lower-case mapping", type, is(MyType.A));
+
+        type = manager.map(config.get("type-A"), MyType.class);
+        assertThat("Upper-case mapping", type, is(MyType.A));
+
+        type = manager.map(config.get("type-special-b"), MyType.class);
+        assertThat("Mixed-clase and hyphen mapping", type, is(MyType.SPECIAL_B));
+
+        type = manager.map(config.get("type-special_b"), MyType.class); // <-- That's an underscore, not a hyphen.
+        assertThat("Mixed-clase and no hyphen mapping", type, is(MyType.SPECIAL_B));
```

Comment:
I think am missing the point of this comment.

Are you suggesting that the config _key_ for a node is somehow linked to the `enum` value names?

The point of `enum` support is to map config _values_ to `enum` values. I chose the config keys in the tests to help me keep track of which key was paired with which values so I could write the test code to check the right conditions.

There is intended to be _no_ correspondence between config _keys_ and the `enum` values.

Reply:
The simpler solution could have been achieved in ~ 3 lines of code in the base config as method:

if (type.isEnum()) {
  Enum.valueOf(type, asString());
}

Obviously this is not capturing the naming heuristics, and not gracefully dealing with exceptions, etc - but that does the heart of what I think we are looking for.

Reply:
1. Please clarify the role you see the config key playing in all this. There is no such role and I did not intend to convey that there is one. If I have suggested that somehow I need to make sure the doc does not do so.
2. Re: a 3-line solution

A quick look at `as` from `ConfigExistingImpl` (_one_ of the implementations of `Config`) and a look at what happens downstream confirms otherwise:
https://github.com/helidon-io/helidon/blob/f4ff8438667b00c43494d9716feec5e15f21b292/config/config/src/main/java/io/helidon/config/ConfigExistingImpl.java#L72-L75

This kicks off the whole config mapping mechanism which includes priorities among mappers, runtime look-ups of additional mapping code, and--importantly--user-provided mappers and mapper providers which can override any of Helidon's built-in ones.